.github/workflows/binderbot.yaml

```diff
@@ -0,0 +1,76 @@
+name: Binderbot
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the master branch
+  push:
+    branches:
+      - master
+    paths:
+      - 'examples/*.ipynb'
+      - 'binder-gallery.yaml'
+      - 'doc/source/_static/icepyx_logo.png'
+  pull_request:
+    branches:
+      - master
+    paths:
+      - 'examples/*.ipynb'
+      - 'binder-gallery.yaml'
+      - 'doc/source/_static/icepyx_logo.png'
+
+jobs:
+  build:
+    name: Binderbot Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: parse config
+        id: parse
+        uses: CumulusDS/get-yaml-paths-action@v0.0.3
+        with:
+          file: binder-gallery.yaml
+          gallery_repository: gallery_repository
+          binder_url: binder_url
+          binder_repo: binder_repo
+          binder_ref: binder_ref
+          binderbot_target_branch: binderbot_target_branch
+      - name: Verify output
+        run: |
+          echo gallery_repository: "${{ steps.parse.outputs.gallery_repository }}"
+          echo binder_url: "${{ steps.parse.outputs.binder_url }}"
+          echo binder_repo: "${{ steps.parse.outputs.binder_repo }}"
+          echo binder_ref: "${{ steps.parse.outputs.binder_ref }}"
+          echo binderbot_target_branch: "${{ steps.parse.outputs.binderbot_target_branch }}"
+      - uses: actions/setup-python@v1
+        with:
+          python-version: '3.8'
+          architecture: 'x64'
+      - name: Install Binderbot from github master
+        run: pip install git+https://github.com/pangeo-gallery/binderbot.git
+      - name: Create Clean Branch
+        run: |
+          git checkout --orphan ${{ steps.parse.outputs.binderbot_target_branch }}
+          git rm --cached -r .
+      - name: Build notebooks using Binderbot
+        run: |
+          python -m binderbot.cli \
+          --binder-url ${{ steps.parse.outputs.binder_url }} \
+          --repo ${{ steps.parse.outputs.binder_repo }} \
+          --ref ${{ steps.parse.outputs.binder_ref }} \
+          examples/*.ipynb
+      - name: Commit files
+        if: github.ref == 'refs/heads/master'
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git add binder-gallery.yaml doc/source/_static/icepyx_logo.png examples/*.ipynb
+          git commit -m "Binderbot output"
+      - name: Push commit
+        if: github.ref == 'refs/heads/master'
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: ${{ steps.parse.outputs.binderbot_target_branch }}
+          force: true
+      - name: Trigger repository dispatch on Pangeo Gallery
+        if: github.ref == 'refs/heads/master'
+        run: curl -f https://pangeo-gallery-bot.herokuapp.com/gallery/submodule-dispatch/${{ github.repository }}
```

Comment:
With the `if: github.ref == 'refs/heads/master'` line(s), it seems that the commit-push-deploy will only happen on the master branch (which is good). Would it make sense then to run this binderbot GIthub Action on Pull Requests targeting both the master and development branches then, so we can test that it works on development too? (See suggestion on other comment).

> > Would it make sense to add this as part of the test suite (currently done with Travis CI)? It tests both the branch and the PR merge.
> 
> I have never really messed with Travis CI, I could probably do it with GitHub Actions though. If you know a bit about Travis, does it seem viable to take the Action in [`binderbot.yaml`](https://github.com/icesat2py/icepyx/pull/125/files#diff-4c6ebb5b85d8a35a622252655a0edd44) and get it in Travis?

It should be fine having both Travis CI and Github Actions CI, a rewrite won't be needed. Travis will run the `icepyx` unit tests as usual and Github Actions will 'test' that the example gallery can be built correctly.

Reply:
Aha! I think you are right @weiji14 . It should only run those last three on `master`, so we might have everything we need if we just add `development` to the branches for the Action to listen to.